### PR TITLE
feat(subscriptions): integrate quarterly billing period

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

Main point of this feature is to add quarterly billing interval

## Description

In this PR there are few main parts:

- updated billing service
- add new interval in plan model
- connect quarterly_billing_service into dates_service

Other parts include tests and configuration

### CODE REVIEW

File: spec/services/invoices/calculate_fees_service_spec.rb [hunk 1]
```diff
@@ -655,6 +655,120 @@
...
+
+        context 'when plan is pay in advance' do
+          let(:pay_in_advance) { true }
+          let(:invoice_subscription) { create(:invoice_subscription, invoice: old_invoice, subscription:) }
+          let(:old_invoice) do
+            create(
+              :invoice,
+              created_at: started_at - 3.months,
+              customer: subscription.customer,
+              organization: plan.organization
```
Comment (jdenquin) [inline spec/services/invoices/calculate_fees_service_spec.rb:717]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

File: app/services/subscriptions/billing_service.rb [hunk 2]
```diff
@@ -109,6 +113,28 @@ def monthly_calendar
...
     end
 
+    # NOTE: Billed quarterly on 1st day of the January, April, July and October
+    def quarterly_calendar
+      billing_month = <<-SQL
+        (
+          DATE_PART('month', (:today#{at_time_zone})) = 1
+            OR DATE_PART('month', (:today#{at_time_zone})) = 4
+            OR DATE_PART('month', (:today#{at_time_zone})) = 7
+            OR DATE_PART('month', (:today#{at_time_zone})) = 10
```
Comment (vincent-pochet) [inline app/services/subscriptions/billing_service.rb:123]: I think you can make it simpler with:
```SQL
DATE_PART('month', (:today#{at_time_zone})) IN (1, 4, 7, 10)
```

File: app/services/subscriptions/billing_service.rb [hunk 3]
```diff
@@ -164,6 +190,47 @@ def monthly_anniversary
...
+          END
+        )
+      SQL
+
+      billing_month = <<-SQL
+        (
+          -- We need to avoid zero and instead of it use 12. E.g.: (3 + 9) % 12 = 0 -> 12
+          CASE WHEN MOD(CAST(DATE_PART('month', (subscriptions.subscription_at#{at_time_zone})) AS INTEGER), 3) = 0
+          THEN
+            (DATE_PART('month', :today#{at_time_zone}) IN (SELECT unnest(ARRAY[3, 6, 9, 12])))
```
Comment (vincent-pochet) [inline app/services/subscriptions/billing_service.rb:215]: Not sure about the SELECT UNNEST, the `IN` operator with a list is not enough?
```suggestion
            (DATE_PART('month', :today#{at_time_zone}) IN (3, 6, 9, 12)
```

**Review by vincent-pochet — APPROVED** [review]
Two small comments regarding the SQL queries, nothing major, it looks great!!

File: app/services/subscriptions/billing_service.rb (hunk 3 above)
Comment (lovrocolic) [reply app/services/subscriptions/billing_service.rb]: It hasn't worked without unnest but there is a chance that I was using wrong brackets 😅  Working fine with new code 👍

**Pull request merged**